libraries/heft-config-file/src/ConfigurationFile.ts

```diff
@@ -191,9 +191,10 @@ export class ConfigurationFile<TConfigurationFile> {
     return this.__schema;
   }
 
-  private readonly _configurationFileCache: Map<string, IConfigurationFileCacheEntry<TConfigurationFile>> =
-    new Map<string, IConfigurationFileCacheEntry<TConfigurationFile>>();
-  private readonly _fileExistsCache: Map<string, boolean> = new Map<string, boolean>();
+  private readonly _configPromiseCache: Map<
+    string,
+    Promise<IConfigurationFileCacheEntry<TConfigurationFile>>
+  > = new Map<string, Promise<IConfigurationFileCacheEntry<TConfigurationFile>>>();
```

Comment:
Pretty sure you should just be able to use `new Map()`; the type is already fully specified on the line, so I don't think you'll hit a lint error?

Reply:
+1  This is more readable and completely cool in the type system:

`_configPromiseCache: Map<string, Promise<IConfigurationFileCacheEntry<TConfigurationFile>>> = new Map();`